dev.zed.Zed-Preview.yaml

```diff
@@ -0,0 +1,208 @@
+        url: https://libbsd.freedesktop.org/releases/libbsd-0.12.2.tar.xz
+        sha256: b88cc9163d0c652aaf39a99991d974ddba1c3a9711db8f1b5838af2a14731014
+        x-checker-data:
+          type: anitya
+          project-id: 1567
+          stable-only: true
+          url-template: https://libbsd.freedesktop.org/releases/libbsd-$version.tar.xz
+    cleanup:
+      - /lib/pkgconfig
+      - /share/man
+
+  - name: netcat
+    buildsystem: simple
+    build-commands:
+      - |
+        while read patch; do
+          echo "Applying $patch..."
+          patch -Np1 -i "debian/patches/$patch"
+        done < debian/patches/series
+      - make CFLAGS="$CFLAGS -I/usr/include/libbsd" LDFLAGS="$LDFLAGS -lbsd"
+      - install -Dm0755 nc /app/bin/nc
+    sources:
+      - type: archive
+        url: https://salsa.debian.org/debian/netcat-openbsd/-/archive/debian/1.228-1/netcat-openbsd-debian-1.228-1.tar.gz
+        sha256: a1eb2593335123efc6e9c12a7e73c72cbf0b665739130fc383322f36757f37cc
+        x-checker-data:
+          type: anitya
+          project-id: 21535
+          stable-only: true
+          url-template: https://salsa.debian.org/debian/netcat-openbsd/-/archive/debian/$version-1/netcat-openbsd-debian-$version-1.tar.gz
+
+  - name: zed-preview
+    buildsystem: simple
+    build-commands:
+      - install -Dm 755 bin/* --target-directory ${FLATPAK_DEST}/bin
+      - install -Dm 755 lib/* --target-directory ${FLATPAK_DEST}/lib
+      - install -Dm 755 libexec/* --target-directory ${FLATPAK_DEST}/libexec
+      - install -Dm 644 share/applications/* --target-directory ${FLATPAK_DEST}/share/applications
+      - install -Dm 644 ${FLATPAK_ID}.metainfo.xml --target-directory ${FLATPAK_DEST}/share/metainfo
+      - install -Dm 644 share/icons/hicolor/512x512/apps/zed.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/zed-preview.png
+
+      # Rename instances of `zed` to `${FLATPAK_ID}`
+      - rename zed-preview ${FLATPAK_ID} ${FLATPAK_DEST}/share/{applications/*,icons/hicolor/*/apps/*}
+
+      # Ensure `desktop-file-edit` validates; Enforce `[Desktop Action NewWorkspace]` has a matching action
+      - |
+        if ! grep -q "^Actions=NewWorkspace" "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"; then
+          # Find the line number where [Desktop Action NewWorkspace] is located
+          line_num=$(grep -n "^\[Desktop Action NewWorkspace\]" "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop" | cut -d: -f1)
+          if [ -n "$line_num" ]; then
+            # Insert Actions=NewWorkspace above the line where [Desktop Action NewWorkspace] is located
+            sed -i "${line_num}i Actions=NewWorkspace" "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+          else
+            # If [Desktop Action NewWorkspace] doesn't exist, add Actions=NewWorkspace at the end
+            echo "Actions=NewWorkspace" >> "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+          fi
+        fi
+
```

Comment:
how about creating a patch instead of this?

Reply:
thanks I removed it.